### PR TITLE
Feedback categories

### PR DESCRIPTION
**Pull request by toumix**
- [x] add more unit tests to get coverage to 98%
- [x] add documentation for the axioms of feedback categories
  - [x] vanishing and joining come for free
- [x] add documentation for the axioms of streams, i.e. the isomorphism `x == x.head @ x.tail.delay()`, etc.
- [x] add the paper https://arxiv.org/abs/2202.02061 to the bibliography

### CODE REVIEW

File: discopy/feedback.py [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+        """ The delay of a feedback type by `n_steps`. """
+        return type(self)(*tuple(x.delay(n_steps) for x in self.inside))
+
+    def reset(self):
+        return type(self)(*tuple(x.reset() for x in self.inside))
+
+    @property
+    def head(self):
+        """ The head of a feedback type, see :class:`HeadOb`. """
+        return type(self)(*(x.head for x in self.inside if x.head))
```
Comment (y-richie-y) [inline discopy/feedback.py:235]: Why is there no `tuple()`  here vs above?

**Review by y-richie-y — APPROVED** [review]
Nice!

File: discopy/feedback.py (hunk 1 above)
Comment (toumix) [reply discopy/feedback.py:236]: They were redundant above, I removed them!

**Comment by toumix** [comment]
I simplified the `__str__` to make things simpler!

**Pull request merged**